terraform/gcp/README.md

```diff
@@ -0,0 +1,259 @@
+
+Includes:
+
+* Installing HashiCorp Tools (Nomad, Consul, Vault, Terraform, and Packer).
+* Installing the GCP SDK CLI Tools, if you're not using Cloud Shell.
+* Creating a new GCP project, along with a Terraform Service Account.
+* Building a golden image using Packer.
+* Deploying a cluster with Terraform.
+
+## Install HashiCorp Tools
+
+### Nomad
+
+Download the latest version of [Nomad](https://www.nomadproject.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/nomad/0.12.0/nomad_0.12.0_linux_amd64.zip" -o nomad.zip
+unzip nomad.zip
+sudo mv nomad /usr/local/bin
+nomad --version
+```
+
+### Consul
+
+Download the latest version of [Consul](https://www.consul.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/consul/1.8.0/consul_1.8.0_linux_amd64.zip" -o consul.zip
+unzip consul.zip
+sudo mv consul /usr/local/bin
+consul --version
+```
+
+### Vault
+
+Download the latest version of [Vault](https://www.vaultproject.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/vault/1.4.3/vault_1.4.3_linux_amd64.zip" -o vault.zip
+unzip vault.zip
+sudo mv vault /usr/local/bin
+vault --version
+```
+
+### Packer
+
+Download the latest version of [Packer](https://www.packer.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/packer/1.6.0/packer_1.6.0_linux_amd64.zip" -o packer.zip
+unzip packer.zip
+sudo mv packer /usr/local/bin
+packer --version
+```
+
+### Terraform
+
+Download the latest version of [Terraform](https://www.terraform.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/terraform/0.12.28/terraform_0.12.28_linux_amd64.zip" -o terraform.zip
+unzip terraform.zip
+sudo mv terraform /usr/local/bin
+terraform --version
+```
+
+### Install the GCP SDK Command Line Tools
+
+> **Note**: if you are using the free [Google Cloud Shell](https://cloud.google.com/shell) VM, you will already have `gcloud` installed. So, you can safley skip this step.
+
+To install the GCP SDK Command Line Tools, follow the installation instructions for your specific operating system: 
+
+* [Linux](https://cloud.google.com/sdk/docs/downloads-interactive#linux)
+* [MacOS](https://cloud.google.com/sdk/docs/downloads-interactive#mac)
+* [Windows](https://cloud.google.com/sdk/docs/downloads-interactive#windows)
+
+#### Initialize the SDK
+
+To perform common setup tasks like authorizing with GCP using `gcloud`, if you haven't already done so, run the following command:
+
+```console
+gcloud init
+```
+
+## Create a New Project
+
+Create a new GCP project with the following command:
+
+```console
```

Comment:
🤔 Looks like `export GOOGLE_PROJECT="nomad-gcp-$(cat /dev/random | head -c 5 | xxd -p)"` could be a semi-portable way of generating a uniq project ID.

Reply:
Hm, when I ran `gcloud init` a few lines up it prompted me for a project name. The name was then automatically exported as `GOOGLE_CLOUD_PROJECT`. Could we skip this step and use that env var instead?

Reply:
Huh. 
Did you do this on Cloud Shell or locally? 
Did you use an existing project? You do not want to use an existing project

Reply:
IIRC, GOOGLE_CLOUD_PROJECT has baggage and implications we do not want, and should be considered reserved for internal use. We don't want to use it because we want to create a shiny new project that is globally unique for all time (whee), and because it is not guaranteed, but we also need to handle the case of `gcloud init` prompting for it 🤔

Reply:
It make sense to not use `gcloud init` at all, since it might confusingly walk people through creating a project. Maybe just using [`gcloud auth login`](https://cloud.google.com/sdk/gcloud/reference/auth/login) could avoid the confusion entirely?

Reply:
> Did you do this on Cloud Shell or locally?
Cloud Shell (pushed the button)

> Did you use an existing project? You do not want to use an existing project...
New project created by `gcloud init`